tests/storage/lvm_test.py

```diff
@@ -1023,6 +1023,56 @@ def test_vg_create_multiple_devices(tmp_storage):
         assert pv.vg_name == ""
 
 
+@requires_root
+@pytest.mark.root
+def test_vg_remove_by_uuid(tmp_storage):
+    dev_size = 10 * GiB
+    num_vgs = 2
+    devs = [tmp_storage.create_device(dev_size) for _ in range(0, num_vgs)]
+    vg_names = [str(uuid.uuid4()) for _ in range(0, num_vgs)]
+
+    # TODO: should work also in read-only mode.
+    for idx, vg_name in enumerate(vg_names):
+        lvm.createVG(vg_name, [devs[idx]], f"initial-tag-{idx}", 128)
+
+    clear_stats()
+    vg = lvm.getVG(vg_names[0])
+    check_stats(hits=0, misses=1)
+
+    assert vg.name == vg_names[0]
+    assert vg.pv_name == (devs[0],)
+    assert vg.tags == ("initial-tag-0",)
+    assert int(vg.extent_size) == 128 * MiB
+
+    lvm.removeVGbyUUID(vg.uuid)
+    # Ensure we have removed the matching VG and only it.
+    with pytest.raises(se.VolumeGroupDoesNotExist):
+        lvm.getVGbyUUID(vg.uuid)
+
+    assert len(lvm.getAllVGs()) == num_vgs - 1
+    for idx, vg_name in enumerate(vg_names[1:], 1):
+        clear_stats()
+        vg = lvm.getVG(vg_name)
+        check_stats(hits=1, misses=0)
+
+        assert vg.name == vg_name
+        assert vg.pv_name == (devs[idx],)
+        assert vg.tags == (f"initial-tag-{idx}",)
+        assert int(vg.extent_size) == 128 * MiB
+
+
+def test_vg_get_invalid_output(monkeypatch, fake_devices):
+    fake_runner = FakeRunner(out=b"Fake lvm output")
+    lc = lvm.LVMCache(fake_runner)
+
+    monkeypatch.setattr(lvm, "_lvminfo", lc)
+
+    with pytest.raises(lvm.InvalidOutputLine) as e:
+        lvm.getVG("vg-name")
+    assert "vgs" in e.value.command
+    assert "Fake lvm output" in e.value.line
```

Comment:
Uhm, cache is empty, I will fill it with some data and think about what would happen to test it properly.

I'd say it shall remain unchanged, except if simulate the fake output with an error, then it might downgrade the stale VGs to `Unreadable`.